### PR TITLE
Support Doctrine DBAL V3

### LINKED ISSUE #325: [Improvement]: Upgrade to Doctrine DBAL v3
### Improvement description

Ideally support both v2 and v3.

### PR DESCRIPTION
**Pull request by blankse**
Resolve https://github.com/pimcore/customer-data-framework/issues/325

### CODE REVIEW

**Comment by kingjia90** [comment]
Thank you, i've tried with 3.4 and seems all working fine thanks to https://github.com/pimcore/customer-data-framework/pull/320 and 326.

Just one thing, `fetchAssoc()` is removed since DBAL [3.2](https://github.com/doctrine/dbal/blob/3.4.x/UPGRADE.md#removed-fetchmode-and-the-corresponding-methods) (point 5) but there are still 2 occurences there 
https://github.com/pimcore/customer-data-framework/blob/f028d4ce886814b9c1f8418c235fc08f710da38c/src/Model/CustomerView/FilterDefinition/Dao.php#L60-L72
this is when pressing `apply filters` in `/admin/customermanagementframework/customers/list`

**Comment by blankse** [comment]
@kingjia90 I changed it.

File: src/Model/ActivityList/DefaultMariaDbActivityList/MariaDbDao.php [hunk 1]
```diff
@@ -100,7 +100,7 @@ private function addLimit(QueryBuilder $select)
     public function getCount()
     {
         $query = $this->getQueryBuilder();
-        $query->setFirstResult(null);
+        $query->setFirstResult(0);
```
Comment (kingjia90) [inline src/Model/ActivityList/DefaultMariaDbActivityList/MariaDbDao.php:103]: @blankse Thank you for the fixes! Looks all good.
I've only noticed that the same change for `MySqlDao.php` would be missing in this PR.
https://github.com/pimcore/customer-data-framework/blob/35a4f04c638321ba86732296e91b7419cf461fff/src/Model/ActivityList/DefaultMariaDbActivityList/MySqlDao.php#L106
(although i see that both are covered in #311 in the minor release branch)

Comment (blankse) [reply src/Model/ActivityList/DefaultMariaDbActivityList/MariaDbDao.php:103]: Done in https://github.com/pimcore/customer-data-framework/pull/327/commits/dbe349cb54e0728ac8465787d51f9cc32c9da4a1

**Pull request merged**